app/controllers/links_controller.rb

```diff
@@ -0,0 +1,30 @@
+require 'uri/http'
+
+class LinksController < ApplicationController
+  def index
+    @link = Link.new
+  end
+
+  def create
+    long_url = params[:link][:long_url]
+    user_chosen_short_url = params[:link][:short_url]
+    @link = CreateLink.short_link_from_existing_link(long_url, user_chosen_short_url)
+    if @link.save
+      flash.now[:notice] = "Your link was shortened to: #{@link.short_url}"
+      redirect_to action: "show", id: @link.id # why not @link??
```

Comment:
You can probably do `redirect_to @link` here. This will do the same as `redirect_to link_path(@link)`.